src/test/java/org/openrewrite/java/migrate/javaee/JavaEE7OpenJPATests.java

```diff
@@ -0,0 +1,231 @@
+              import javax.persistence.GenerationType;
+              import javax.persistence.Id;
+              import javax.persistence.TableGenerator;
+                                     
+              @Entity
+              public class GeneratedValueExample  {
+                                     
+                  // flag it
+                  @Id
+                  @javax.persistence.TableGenerator(name = "OPENJPA_SEQUENCE_TABLE", table = "OPENJPA_SEQUENCE_TABLE", pkColumnName = "ID", valueColumnName = "SEQUENCE_VALUE", pkColumnValue = "0")
+                  @GeneratedValue(strategy = javax.persistence.GenerationType.TABLE, generator = "OPENJPA_SEQUENCE_TABLE")
+                  private int id;
+                                     
+                  // flag it.  Does not require @Id
+                  @GeneratedValue(strategy = javax.persistence.GenerationType.TABLE, generator = "OPENJPA_SEQUENCE_TABLE")
+                  @javax.persistence.TableGenerator(name = "OPENJPA_SEQUENCE_TABLE", table = "OPENJPA_SEQUENCE_TABLE", pkColumnName = "ID", valueColumnName = "SEQUENCE_VALUE", pkColumnValue = "0")
+                  private int id2;
+                                     
+                  // flag it even though it has a TableGenerator since GeneratedValue is default
+                  // A second TableGenerator will be created.
+                  @TableGenerator(name = "SOME_TABLE", table = "SOME_TABLE", pkColumnName = "ID", valueColumnName = "SEQUENCE_VALUE", pkColumnValue = "0")
+                  @javax.persistence.TableGenerator(name = "OPENJPA_SEQUENCE_TABLE", table = "OPENJPA_SEQUENCE_TABLE", pkColumnName = "ID", valueColumnName = "SEQUENCE_VALUE", pkColumnValue = "0")
+                  @GeneratedValue(strategy = javax.persistence.GenerationType.TABLE, generator = "OPENJPA_SEQUENCE_TABLE")
+                  private int id3;
+                                     
+              }
+              """
+          )
+        );
+    }
+
+    @Test
+    void generatedValueName() {
+        rewriteRun(
+          //language=java
+          java(
+            """
+               package com.ibm.test;
+               
+               import javax.persistence.Entity;
+               import javax.persistence.GeneratedValue;
+               import javax.persistence.Id;
+               
+               @Entity
+               public class GeneratedValueName  {
+               
+               	// flag it
+               	@Id
+               	@GeneratedValue
+               	private String name;
+               	
+               
+               }
+               """, """
+                       package com.ibm.test;
+                       
+                       import javax.persistence.Entity;
+                       import javax.persistence.GeneratedValue;
+                       import javax.persistence.Id;
+                       
+                       @Entity
+                       public class GeneratedValueName  {
```

Comment:
Similarly any modifiers that are not needed detract from showing what change the recipe makes. Let's remove those where we can.
```suggestion
                       class GeneratedValueName  {
```